client/components/TestQueue/index.jsx

```diff
@@ -1,329 +1,255 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { useQuery } from '@apollo/client';
 import { Link } from 'react-router-dom';
 import { Container, Table, Alert } from 'react-bootstrap';
 import { Helmet } from 'react-helmet';
-import { connect } from 'react-redux';
-import { getAllUsers } from '../../actions/users';
-import { getActiveRuns, getActiveRunConfiguration } from '../../actions/runs';
-import { handleGetValidAts } from '../../actions/ats';
 import nextId from 'react-id-generator';
-import TestQueueRun from '@components/TestQueueRun';
-import DeleteResultsModal from '@components/DeleteResultsModal';
-import CurrentGitCommit from '@components/CurrentGitCommit';
+import TestQueueRun from '../TestQueueRun';
+import {
+    AddTestPlanToQueueContainer,
+    AddTestPlanToQueueModal
+} from '../AddTestPlanToQueue';
+import DeleteResultsModal from '../DeleteResultsModal';
+
+import { TEST_QUEUE_PAGE_QUERY } from './queries';
 import './TestQueue.css';
 
-class TestQueue extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            showDeleteResultsModal: false,
-            resultsDelete: {
-                username: '',
-                run: null
-            },
-            deleteFunction: null
-        };
-
-        this.deleteResults = this.deleteResults.bind(this);
-        this.showDeleteResultsModal = this.showDeleteResultsModal.bind(this);
-        this.closeDeleteResultsModal = this.closeDeleteResultsModal.bind(this);
-    }
-
-    componentDidMount() {
-        const {
-            dispatch,
-            activeRunsById,
-            usersById,
-            ats,
-            activeRunConfiguration
-        } = this.props;
-
-        if (!activeRunsById) {
-            dispatch(getActiveRuns());
-        }
-
-        if (!activeRunConfiguration) {
-            dispatch(getActiveRunConfiguration());
-        }
-
-        if (!ats) {
-            dispatch(handleGetValidAts());
-        }
-
-        if (!Object.keys(usersById).length) {
-            dispatch(getAllUsers());
+const TestQueue = ({ auth }) => {
+    const { loading, data, refetch } = useQuery(TEST_QUEUE_PAGE_QUERY);
+
+    const [testers, setTesters] = useState([]);
+    const [testPlanReports, setTestPlanReports] = useState([]);
+    const [structuredTestPlanTargets, setStructuredTestPlanTargets] = useState(
+        {}
+    );
+    const [deleteResultsDetails, setDeleteResultsDetails] = useState({});
+    const [isShowingDeleteResultsModal, enableDeleteResultsModal] = useState(
```

Comment:
Perhaps the setter function should start with `set`? I guess that's the way it's normally done at least!

Reply:
I typically prepend with `enable` and `toggle` for my boolean hooks, but that's just a me thing 🙂 so I'll go with the common pattern here